athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -186,10 +186,19 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
                 .filter(index -> !index.startsWith("."));
 
         // get all data streams from ES, 1 data stream can contains multiple indices which start with ".ds-xxxxxxxxxx"
-        Stream<String> dataStreams = client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
-                .stream()
-                .map(dataStream -> dataStream.getName());
-
+        Stream<String> dataStreams;
+        try {
+            dataStreams = client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
+                    .stream()
+                    .map(dataStream -> dataStream.getName());
+        }
+        // gracefully exit for AWS ES.
+        // AWS's version of ES doesn't support data stream, it is a feature only available in the Elasticsearch distributed by Elastic itself, licensed under the Elastic License.
+        // https://stackoverflow.com/questions/68006918/aws-elasticsearch-service-and-datastream
+        catch (Exception ex) {
+            logger.warn(String.format("Unable to retrieve datastream or cluster version not support data stream, ignore datastream, message: %s", ex.getMessage()), ex);
```

Comment:
You should pass in strings to the logger the way we do it everywhere else:
```
logger.warn("Exception retrieving data stream. Message: {}", ex.getMessage());
```

We can use the {} as a placeholder and then just pass in the args after. You could also just do:
```
logger.warn("Exception retrieving data stream: {}", ex);
```